Fix builds on glibc >= 2.38

msepga opened the pull request:
`glibc` now defines `strchrnul`, so we add a check to see if `glibc` is being used, and instead import the symbol if so.

Closes #202

msepga committed:
Fix builds on glibc >= 2.38

`glibc` now defines `strchrnul`, so we add a check to see if `glibc` is
being used, and instead import the symbol if so.

msepga committed:
Bump version to 4.2.3

lfittl commented on CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
 
+## 15-4.2.3   2023-07-07
+
+* Fix builds when compiling with `glibc >=  2.38` [#203](https://github.com/pganalyze/libpg_query/pull/203)
```
Would be good to also add a changelog entry for #199, like:

```
* Deparser: Add support for COALESCE and other expressions in LIMIT clause
```

msepga committed:
Include `stdlib.h` instead of `features.h`

`features.h` is not available on all platforms. We can rely on it being
included via `glibc`'s `stdlib.h` header.

lfittl commented on src/postgres/include/pg_config.h:
```diff
@@ -1032,6 +1032,7 @@
 #undef USE_ARMV8_CRC32C
 #undef USE_SSE42_CRC32C_WITH_RUNTIME_CHECK
-#if defined(__FreeBSD__) || defined(__NetBSD__)
+#include <features.h>
```
This doesn't exist for me on macOS - do we actually need this?

lfittl replied:
(resolved after the last commit)

msepga committed:
Add note about PR #199 in 4.2.3 changelog
